src/dmd/expression.d

```diff
@@ -1818,70 +1819,69 @@ extern (C++) final class IntegerExp : Expression
         return value;
     }
 
-    void setInteger(dinteger_t value)
+    private void setInteger(dinteger_t value)
     {
-        this.value = normalize(type.toBasetype().ty, value);
+        this.value = value;
+        normalize();
     }
 
-    static dinteger_t normalize(TY ty, dinteger_t value)
+    void normalize()
```

Comment:
This PR does not remove technical debt, it adds it. `normalize()` was a function that could be pure. It is refactored into a function that is not pure. I strongly object to this PR. Not only does it not do what the title says it does, I don't see a purpose to these changes, and none is offered.

Reply:
I see you did offer a rationale, making `setInteger()` private. I don't see a gain, though, as `normalize()` gets exposed instead. I don't see any improvement.